test/puppetlabs/http/client/sync_plaintext_test.clj

```diff
@@ -236,6 +258,23 @@
             (is (= "Hello, World!" (slurp (:body response))))))
         (.close client)))))
 
+(deftest cookie-support-test
+  (testlogging/with-test-logging
+    (testutils/with-app-with-config app
+      [jetty9/jetty9-service test-cookie-service]
+      {:webserver {:port 10000}}
+      (let [client (HttpAsyncClients/createDefault)
```

Comment:
This looks like it's testing the underlying apache client directly, rather than going through our API.  I think what we probably want is to have one test that tests our Java API for creating a client and making requests on it, and one that tests our Clojure API for the same.

The java API test would look something like this:

https://github.com/puppetlabs/clj-http-client/blob/master/test/puppetlabs/http/client/sync_plaintext_test.clj#L92-L98

The clojure one would probably look like this:

https://github.com/puppetlabs/clj-http-client/blob/master/test/puppetlabs/http/client/sync_plaintext_test.clj#L132-L136

I think you're testing all of the right stuff, but just making sure it works with our APIs seems worthwhile.